README.md

```diff
@@ -157,6 +197,338 @@ the next request either as a hidden form field or via JavaScript async request.
 
 It is recommended to always use this decorator.
 
+## CSRF protection for AJAX/SPA backend API
+
+If you are using a cookie to authenticate your AJAX/SPA, then you do need CSRF protection for the backend API.
+
+### Employing custom request header
+
+In this pattern, AJAX/SPA frontend appends a custom header to API requests that require CSRF protection. No token is needed for this approach. This defense relies on the CORS preflight mechanism which sends an `OPTIONS` request to verify CORS compliance with the destination server. All modern browsers, according to the same-origin policy security model, designate requests with custom headers as "to be preflighted". When the API requires a custom header, you know that the request must have been preflighted if it came from a browser.
+
+The header can be any arbitrary key-value pair, as long as it does not conflict with existing headers. Empty value is also acceptable.
+
+```
+X-CSRF-HEADER=1
+```
+
+When handling the request, the API checks for the existence of this header. If the header does not exist, the backend rejects the request as potential forgery. Employing a custom header allows to reject [simple requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests) that browsers do not designate as "to be preflighted" and permit them to be sent to any origin.
+
+In order to enable CSRF protection you need to add `CsrfHeaderMiddleware` to your `MiddlewareDispatcher` configuration:
+
+>[yiisoft/middleware-dispatcher](https://github.com/yiisoft/middleware-dispatcher) configuration example
```

Comment:
Same as above. Please fix that in other places. Using quotes for referencing packages doesn't fit overall docs style.

Reply:
The idea is to demonstrate all possible ways to configure `Foo` in general and provide example for appropriate yii package.

The pattern is:

-----
or define the `Foo` configuration in the DI container:

>[yiisoft/di](https://github.com/yiisoft/di) configuration example
```
yii/di example
```
-----


From the following pattern, it's not clear that `Foo` can be configured with DI other than yiisoft/di.

-----
or define the `Foo` configuration in the [DI container](https://github.com/yiisoft/di):
```
yii/di example
```
-----

There are no patterns in the readmes for referencing to yii packages. I'm using a quote, because it allows to highlight a string with a link. It can be any, if quote is not suitable here.

Reply:
Quote is definitely not suitable. I'd just use a plain link. If the idea is to highlight that it's Yii3-specific config, use sub-header.